packages/shared-ux/router/impl/route.tsx

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useMemo } from 'react';
+import {
+  // eslint-disable-next-line no-restricted-imports
+  Route as ReactRouterRoute,
+  RouteComponentProps,
+  RouteProps,
+  useRouteMatch,
+} from 'react-router-dom';
+import { useKibanaSharedUX } from './services';
+import { useSharedUXExecutionContext } from './use_execution_context';
+
+/**
+ * This is a wrapper around the react-router-dom Route component that inserts
+ * MatchPropagator in every application route. It helps track all route changes
+ * and send them to the execution context, later used to enrich APM
+ * 'route-change' transactions.
+ */
+export const Route = <T extends {}>({
+  children,
+  component: Component,
+  render,
+  ...rest
+}: RouteProps<string, { [K: string]: string } & T>) => {
+  const component = useMemo(() => {
+    if (!Component) {
+      return undefined;
+    }
+    return (props: RouteComponentProps) => (
+      <>
+        <MatchPropagator />
+        <Component {...props} />
+      </>
+    );
+  }, [Component]);
+
+  if (component) {
+    return <ReactRouterRoute {...rest} component={component} />;
+  }
+  if (render || typeof children === 'function') {
+    const renderFunction = typeof children === 'function' ? children : render;
```

Comment:
Would this work?

```suggestion
const renderFunction = typeof children === 'function' ? children : render;
 if (renderFunction) {
```
With that we could remove the `ts-ignore` inside this condition

Reply:
Thank you @crespocarlos I think it could work, but like described below, this component was just renamed and I wanted to avoid changing the logic in the current component, so that's why I have decided to keep it as it was